### PR TITLE
sort tasks by UID when task index parsing fails

### PR DESCRIPTION
**Pull request by panoswoo**
Resolves [#3386](https://github.com/volcano-sh/volcano/issues/3386)

The tasks of vc-job is created in goroutine, and the CreationTimestamp of tasks is random. So, sorting by UID when task index parsed failed is a more clear and reasonable method.

### CODE REVIEW

**Comment by lowang-bh** [comment]
let's keep comparing createtime.

**Comment by Monokaix** [comment]
Welcome! we should compare creationTime first, if they are equal, then compare uid: )

**Comment by panoswoo** [comment]
> let's keep comparing createtime.

done

**Comment by panoswoo** [comment]
> Welcome! we should compare creationTime first, if they are equal, then compare uid: )

done

File: pkg/controllers/job/helpers/helpers_test.go [hunk 1]
```diff
@@ -17,12 +17,10 @@ limitations under the License.
 package helpers
 
 import (
-	"testing"
-	"time"
-
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
+	"testing"
```
Comment (Monokaix) [inline pkg/controllers/job/helpers/helpers_test.go:22]: plz sort imports.

Comment (panoswoo) [reply pkg/controllers/job/helpers/helpers_test.go]: done

File: pkg/controllers/job/helpers/helpers_test.go [hunk 2]
```diff
@@ -35,23 +35,33 @@ func TestCompareTask(t *testing.T) {
...
-			generateTaskInfo("prod-worker", createTime),
-			generateTaskInfo("prod-worker-3", createTime.Add(time.Hour)),
+			generateTaskInfo("prod-worker", createTime, "pod-1"),
+			generateTaskInfo("prod-worker-3", createTime.Add(time.Hour), "pod-2"),
 			true,
 		},
 		{
-			generateTaskInfo("prod-worker", createTime),
-			generateTaskInfo("prod-worker-3", createTime.Add(-time.Hour)),
+			generateTaskInfo("prod-worker", createTime, "pod-1"),
```
Comment (Monokaix) [inline pkg/controllers/job/helpers/helpers_test.go:53]: We'd better add two test case that task index parsed failed, and one case is two pods with same timestamp, another case is two pods with differenet timestamp.

Comment (panoswoo) [reply pkg/controllers/job/helpers/helpers_test.go:53]: case 3&4: task index parsed failed but two pods with different timestamp
case 5&6: task index parsed failed and two pods with same timestamp

Comment (panoswoo) [reply pkg/controllers/job/helpers/helpers_test.go:53]: @Monokaix hi, is there anything else that needs to be modified?

Comment (Monokaix) [reply pkg/controllers/job/helpers/helpers_test.go:53]: Ok, will trigger the CI later: )

**Review by william-wang — APPROVED** [review]
/approve

**Review by william-wang — COMMENTED** [review]
/lgtm

**Pull request merged**